src/pages/[platform]/build-a-backend/storage/use-with-custom-s3/index.mdx

```diff
@@ -85,28 +92,344 @@ If you specify a custom S3 bucket, no sandbox storage resource will be created.
 
 </Callout>
 
-```javascript
+Below is an example of configuring the backend to define a custom S3 bucket where only authenticated (i.e. signed in) users have full access to a folder called `public/`:
 
-import { defineBackend } from '@aws-amplify/backend';
-import { auth } from './auth/resource';
-import { data } from './data/resource';
+```ts title="amplify/backend.ts"
+import { defineBackend } from "@aws-amplify/backend";
+import { auth } from "./auth/resource";
 
 const backend = defineBackend({
   auth,
-  data,
 });
-
 //highlight-start
 backend.addOutput({
   storage: {
     aws_region: "<region>",
-    bucket_name: "<bucket-name>"
+    bucket_name: "<bucket-name>",
+    // optional: `buckets` can be used when setting up more than one existing bucket
+    buckets: [
+      {
+        aws_region: "<region>",
+        bucket_name: "<bucket-name>",
+        name: "<bucket-name>",
+        // optional: `paths` can be used to set up access to specific bucket prefixes,
+        // and configure user access types to those prefixes
+        paths: {
+          "public/*": {
+            authenticated: ["get", "list", "write", "delete"],
+          },
+        },
+      }
+    ]
+  },
+});
+
+// Define an inline policy to attach to Amplify's auth role
+// This policy defines how authenticated users can access your existing bucket
+const authPolicy = new Policy(backend.stack, "customBucketAuthPolicy", {
+  statements: [
+    new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ["s3:GetObject", "s3:PutObject", "s3:DeleteObject"],
+      resources: ["arn:aws:s3:::<bucket-name>/public/*",],
+    }),
+    new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ["s3:ListBucket"],
+      resources: [
+        "arn:aws:s3:::<bucket-name>",
+        "arn:aws:s3:::<bucket-name>/*"
+        ],
+      conditions: {
+        StringLike: {
+          "s3:prefix": ["public/*", "public/"],
+        },
+      },
+    }),
+  ],
+});
+
+// Add the policies to the authenticated user role
+backend.auth.resources.authenticatedUserIamRole.attachInlinePolicy(authPolicy);
+// highlight-end
+```
+
+From there, you can further configure the backend to add custom authorization rules for different user types. 
+
+<Callout>
+The custom authorization rules defined in the examples below are able to be combined and follow the same rules used when working with Amplify-defined storage. For more information about the access types and access definition rules supported by Amplify, please refer to our documentation on [customizing authorization rules](/[platform]/build-a-backend/storage/authorization/). 
+</Callout>
+
+<BlockSwitcher>
+<Block name="Guest Users">
+Below is an example of expanding the original backend object to grant all guest (i.e. not signed in) users read access to files under `public/`:
+
+```ts title="amplify/backend.ts"
+
+// ...
+
+backend.addOutput({
+  storage: {
+    aws_region: "<region>",
```

Comment:
can we pivot this to `s3.Bucket.fromBucketName()`? this should help with the access control shown later https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.Bucket.html#static-fromwbrbucketwbrnamescope-id-bucketname